src/adapters/encounter-diagnoses-adapter.ts

```diff
@@ -0,0 +1,80 @@
+import { type OpenmrsResource } from '@openmrs/esm-framework';
+import { type FormFieldValueAdapter, type FormProcessorContextProps } from '..';
+import { type FormContextProps } from '../provider/form-provider';
+import { type OpenmrsEncounter, type FormField } from '../types';
+import { clearSubmission, gracefullySetSubmission } from '../utils/common-utils';
+
+export let assignedDiagnosesIds: string[] = [];
+
+export const EncounterDiagnosesAdapter: FormFieldValueAdapter = {
```

Comment:
```suggestion
export const EncounterDiagnosisAdapter: FormFieldValueAdapter = {
```

Reply:
We agreed to name it in plural since it can be more than one [here](https://github.com/openmrs/openmrs-esm-form-engine-lib/pull/298#discussion_r1775106522)

Reply:
Plural makes sense for the variables because you are actually dealing with multiple but singular makes more sense for file names and for this case the function/class. Kinda the same way we have `EncounterLocationAdapter` and `EncounterRoleAdapter`

Reply:
This is more similar to orders from what i understood, and its in plural, @samuelmale can also weigh in since it was part of his review